src/lib/multiaddres.ts

```diff
@@ -279,33 +289,37 @@ function getRandomRelayAddr(
   return r;
 }
 
-export function resolveRelay(
-  maybeRelay: string | undefined,
-  fluenceEnv: FluenceEnv,
-  maybeFluenceConfigReadonly: FluenceConfigReadonly | null,
-) {
-  return (
-    getMaybeNamedNode(maybeRelay, maybeFluenceConfigReadonly)?.multiaddr ??
-    maybeRelay ??
-    getRandomRelayAddr(fluenceEnv, maybeFluenceConfigReadonly)
-  );
+export async function resolveRelay({
+  maybeRelay,
+  ...args
+}: { maybeRelay: string | undefined } & ResolveNodesArgs) {
+  const namedAddr = (
+    await getMaybeNamedAddrAndPeerId(maybeRelay, args.maybeFluenceConfig)
+  )?.multiaddr;
+
+  if (namedAddr !== undefined) {
+    return namedAddr;
+  }
+
+  if (maybeRelay !== undefined) {
+    return maybeRelay;
+  }
+
+  return getRandomRelayAddr(args);
```

Comment:
```suggestion
return namedAddr ?? maybeRelay ?? getRandomRelayAddr(args);
```

Reply:
yeah I just recently changed that so it's not so much nullish stuff, because folex asked me to. But if I move away the await outside like I did - nullish stuff doesn't look that bad anymore, that's true